components/validator/issuer.go

```diff
@@ -35,17 +36,23 @@ func issueValidatorBlock(ctx context.Context) {
 		return
 	}
 
-	modelBlock, err := deps.BlockIssuer.CreateValidationBlock(ctx,
-		validatorAccount,
-		blockfactory.WithValidationBlockHeaderOptions(
-			blockfactory.WithIssuingTime(blockIssuingTime),
-			blockfactory.WithSlotCommitment(latestCommitment.Commitment()),
-		),
-		blockfactory.WithProtocolParametersHash(protocolParametersHash),
-		blockfactory.WithHighestSupportedVersion(deps.Protocol.LatestAPI().Version()),
-	)
+	// create the validation block here using the validation block builder from iota.go
+
+	validationBlock, err := builder.NewValidationBlockBuilder(deps.Protocol.CurrentAPI()).
+		IssuingTime(blockIssuingTime).
+		SlotCommitmentID(latestCommitment.ID()).
+		ProtocolParametersHash(protocolParametersHash).
+		HighestSupportedVersion(deps.Protocol.LatestAPI().Version()).
+		Build()
```

Comment:
I am missing `LatestFinalizedSlot`, the issuer and the signing of the block here

Reply:
Also tipselection is missing

Reply:
Added basic tip selection here as I don't think we have special validator tip selection yet.